app/views/govuk_publishing_components/components/_layout_for_public.html.erb

```diff
@@ -7,6 +7,26 @@
   omit_header ||= false
   show_search = local_assigns.include?(:show_search) ? local_assigns[:show_search] : true
   title ||= "GOV.UK - The best place to find government services and information"
+
+# This is a hack - but it's the only way I can find to not have two blue bars on
+# constrained width layouts.
+#
+# The full width layout hides the blue bar underneath the blar header bar - so
+# full width pages won't see the blue bar unless it's added by another component
+# - and for most pages that component is the global banner.
+#
+# The constrained width layout doesn't hide the blue bar - so having the global
+# banner on a constrained width layout means there are two blue bars.
+#
+# The global banner is shown with JavaScript, so users without JavaScript won't
+# see it. So the constrained width blue bar can't be turned off as then it'll be
+# off for everyone.
+#
+# This booleon adds a CSS class that shifts the banners up by the blue bar's
+# height, making the two blue bars overlap and appear as one. The class is added
+# when a) there's content for the emergency or global banner *and* b) when using
+# the contrained width layout.
+  blue_bar_dedupe = !full_width && global_bar.present?
```

Comment:
As discussed, I think we can afford this bit of depth to be able to move on with the main objective. The global bar design is likely to change soon and I'd rather try to address it when iterating on it.